README.md

```diff
@@ -10,6 +10,53 @@ official packaging.
 
 ## Usage
 
+### Documentation
+
+The contents of `default.nix` is a function from a set of common nix packages
+to a set of functions. `default.nix` is meant to be called via the callPackage
+function. The functions it produces are as follows.
+
+#### rustc
+
+`rustc` is a function from an optional `date` (defaults to latest), and
+an optional `system` (defaults to the system nix is installed on) to
+a derivation for the rust compiler. If you wish, you can provide an
+optional hash argument to specify exactly what the blob that 
+rust-nightly-nix downloads should hash to. If no hash is provided,
+rust-nightly-nix will just trust whatever it gets.
```

Comment:
Actually it will blindly trust the hash mozilla provides (i.e. that blob is unhashed) and then use that the verify the actual download.